ibis/backends/postgres/registry.py

```diff
@@ -404,36 +397,40 @@ def _mod(t, op):
 
 
 def _neg_idx_to_pos(array, idx):
-    return sa.case(
-        [
-            (array.is_(None), None),
-            (idx < 0, sa.func.array_length(array, 1) + idx),
-        ],
-        else_=idx,
-    )
+    return sa.case([(idx < 0, sa.func.cardinality(array) + idx)], else_=idx)
 
 
-def _array_slice(t, op):
-    arg, start, stop = op.args
-    sa_arg = t.translate(arg)
-    sa_start = t.translate(start)
+def _array_slice(*, index_converter, array_length):
```

Comment:
Why not to use `_neg_idx_to_pos` here directly?

Reply:
Just so that the code can be shared with the duckdb implementation.